tests/test-users-api.js

```diff
@@ -0,0 +1,236 @@
+            [birthdate, signupdate, name, cmnd, email, gender, expirydate])
+            new_user_id = result.rows[0].new_id + '1';
+            console.log("id: " + new_user_id)
+            await axios.get( `${process.env.API_URL}/users/${new_user_id}`).catch(error=>{
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+            })
+            await pool.query('CALL public."DeleteUser" ($1)', [result.rows[0].new_id])
+        })
+        it('4. should return status 500 and response will be null when called with incorrect id\'s value', async() => {
+            var new_user_id;
+            const result = await pool.query('SELECT public."InsertUser"($1, $2, $3, $4, $5, $6, $7) as new_id', 
+            [birthdate, signupdate, name, cmnd, email, gender, expirydate])
+            new_user_id = result.rows[0].new_id.substring(0,36) + '1'
+            console.log("id: " + new_user_id)
+            await axios.get( `${process.env.API_URL}/users/${new_user_id}`).catch(error=>{
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+            })
+            await pool.query('CALL public."DeleteUser" ($1)', [result.rows[0].new_id])
+        })
+    })
+
+    describe('create user', async() => {
+        it('5. should return status 201 and have a different id when called', async() => {
+            const response = await axios.post(`${process.env.API_URL}/users`, {
+                name: name,
+                gender: gender,
+                cmnd: cmnd,
+                email: email,
+                birthdate: birthdate.toLocaleDateString(),
+                signupdate: signupdate.toLocaleDateString(),
+                expirydate: expirydate.toLocaleDateString()
+            })
+            console.log('id khuong: ' + response.data.InsertedUserId)
+            const result = pool.query('SELECT * FROM users WHERE id = $1', [response.data.InsertedUserId], (err, result) => {
+                if(err) {
+                    throw err;
+                }
+                const dbUser = result.rows[0];
+                assert.strictEqual(dbUser.name, name);
+                assert.strictEqual(dbUser.cmnd, cmnd);
+                assert.strictEqual(dbUser.gender, gender);
+                assert.strictEqual(dbUser.birthdate.toISOString(), birthdate.toISOString());
+                assert.strictEqual(dbUser.email, email);
+                assert.strictEqual(dbUser.signupdate.toISOString(), signupdate.toISOString());
+                assert.strictEqual(dbUser.expirydate.toISOString(), expirydate.toISOString());
+            })
+            assert.strictEqual(response.status, 201)
```

Comment:
Should remove inserted user after completed test case

Reply:
I fixed it